#!/usr/bin/env python
PACKAGE = "mbf_rrts_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("iterations",    int_t,    0, "Number of iterations to run for RRT*", 1500,  1)
gen.add("seed",    int_t,    0, "Number of iterations to run for RRT*", 123)
gen.add("neighbourhood_size", double_t, 0, "Side length of the square neighbourhood around a node",    4.0)
gen.add("distance_factor", double_t, 0, "The multiplying factor for the distance cost between two nodes", 7.0)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

# size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                        gen.const("Medium",     int_t, 1, "A medium constant"),
#                        gen.const("Large",      int_t, 2, "A large constant"),
#                        gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                      "An enum to set size")

# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "mbf_rrts_planner", "mbf_rrts_planner"))